File: longest-substring-without-repeating-characters.js
function lengthOfLongestSubstring(s) {
    var left = 0;
    var longest = 0;
    var subString = new Set();
    for (var _i = 0, s_1 = s; _i < s_1.length; _i++) {
        var right = s_1[_i];
        while (subString.has(right)) {
            subString.delete(s[left]);
            left += 1;
        }
        subString.add(right);
        longest = Math.max(longest, subString.size);
    }
    return longest;
}
;
console.log(lengthOfLongestSubstring("abcabcbb"));
console.log(lengthOfLongestSubstring("bbbbb"));
console.log(lengthOfLongestSubstring("pwwkew"));

File: longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        subStr = set()
        left = 0
        longest = 0

        for right in range(len(s)):
            while s[right] in subStr:
                subStr.remove(s[left])
                left += 1
            subStr.add(s[right])
            longest = max(longest, right - left + 1)
        return longest

solution = Solution()

print(solution.lengthOfLongestSubstring("abcabcbb"))  #3
print(solution.lengthOfLongestSubstring("bbbbb"))  #1
print(solution.lengthOfLongestSubstring("pwwkew")) #3